packages/react-dom/src/__tests__/ReactDOMFizzServer-test.js

```diff
@@ -5,6 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  *
  * @emails react-core
+ * @jest-environment ./scripts/jest/ReactDOMServerIntegrationEnvironment
```

Comment:
In Jest v26, the module path 'react-dom/server' would map to the file path 'packages/react-dom/server.js'. However, in Jest v29, this mapping rule has changed and it will first look at the exports field of the package.json file, and then decide which file to load based on the testing environment. In jest-environment-jsdom, 'packages/react-dom/server.browser.js' will be loaded, and in jest-environment-node, 'packages/react-dom/server.node.js' will be loaded. This will cause tests that use node environment methods (such as renderToPipeableStream) to fail. Is it possible to use the module path 'react-dom/server.node'? I see that the problem can also be solved by customizing the testing environment (https://github.com/facebook/react/pull/21575), adding some jsdom environment variables in the node testing environment.